injected/src/features/ua-ch-brands.js

```diff
@@ -0,0 +1,130 @@
+import ContentFeature from '../content-feature';
+
+export default class UaChBrands extends ContentFeature {
+    constructor(featureName, importConfig, args) {
+        super(featureName, importConfig, args);
+
+        this.cachedBrands = null;
+        this.originalBrands = null;
+    }
+
+    init() {
+        const configuredBrands = this.getFeatureSetting('brands');
+        this.log.info(
+            'init() - configured brands from settings:',
+            configuredBrands ? configuredBrands.map((b) => `"${b.brand}" v${b.version}`).join(', ') : 'null',
+        );
+
+        if (!configuredBrands || configuredBrands.length === 0) {
+            this.log.info('No client hint brands correctly configured, feature disabled');
+            return;
+        }
+        this.shimUserAgentDataBrands();
+    }
+
+    /**
+     * Override navigator.userAgentData.brands to match the Sec-CH-UA header
+     */
+    shimUserAgentDataBrands() {
+        try {
+            // @ts-expect-error - userAgentData not yet standard
+            if (!navigator.userAgentData || !navigator.userAgentData.brands) {
+                this.log.info('shimUserAgentDataBrands - navigator.userAgentData not available');
+                return;
+            }
+
+            if (!this.originalBrands) {
+                // @ts-expect-error - userAgentData not yet standard
+                this.originalBrands = [...navigator.userAgentData.brands];
+                this.log.info('shimUserAgentDataBrands - captured original brands:', this.originalBrands.map(b => `"${b.brand}" v${b.version}`).join(', '));
+            }
+
+            if (this.cachedBrands) {
```

Comment:
I don't even think we need this check do we? (and the `cachedBrands` prop).

`shimUserAgentDataBrands` is called once in `init` which is also called once.